code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesLocationManager.java

```diff
@@ -254,58 +257,35 @@ void setLocationPermission(PlacesMonitorLocationPermission placesMonitorLocation
 
 	/**
 	 * Handler for processing the received location event.
-	 *
 	 * <p>
-	 * This method is called by the internal {@link android.content.BroadcastReceiver} on receiving an intent with {@link Location}.
-	 * Calls the {@link PlacesExtension} to get the closest POIs around the given location.
-	 *
-	 * No action is performed if the intents action is not same as {@link PlacesMonitorConstants#INTERNAL_INTENT_ACTION_LOCATION}.
-	 * No action is performed if the received {@code LocationResult} is null.
-	 * No action is performed if the location array or the location is null.
+	 * This method will be called when the OS event on location update is received.
+	 * This method attempts to fetch and monitor 20 near by POIs around the given location.
 	 *
-	 * @param intent broadcasted geofence event message wrapped in an intent
+	 * @param eventData {@link EventData} from the location update OS event.
 	 * @see Places#getNearbyPointsOfInterest(Location, int, AdobeCallback)
 	 */
-	void onLocationReceived(final Intent intent) {
-
-		if (intent == null) {
-			Log.warning(PlacesMonitorConstants.LOG_TAG,
-						"Cannot process the location update, The received intent from the location broadcast receiver  is null");
-			return;
+	void onLocationReceived(final EventData eventData) {
+		double latitude;
+		double longitude;
+		try {
+			latitude = eventData.getDouble(PlacesMonitorConstants.EventDataKey.LATITUDE);
+			longitude = eventData.getDouble(PlacesMonitorConstants.EventDataKey.LONGITUDE);
 		}
-
-		final String action = intent.getAction();
-
-		if (!PlacesMonitorConstants.INTERNAL_INTENT_ACTION_LOCATION.equals(action)) {
-			Log.trace(PlacesMonitorConstants.LOG_TAG,
-					  "Cannot process the location update, Invalid action type received from location broadcast receiver");
+		catch (VariantException exception) {
+			Log.warning(PlacesMonitorConstants.LOG_TAG, "PlacesLocationManager : Exception occurred while extracting latitude/longitude from the OS event. Ignoring location update event." + exception.getMessage());
```

Comment:
Same as comment above with message format

Reply:
good point 👍. I changed them in everyplace for readability